core/capabilities/syncer.go

```diff
@@ -56,26 +71,486 @@ var defaultStreamConfig = p2ptypes.StreamConfig{
+					s.lggr,
+				)
+				return client, nil
+			}
+
+			err := s.addToRegistryAndSetDispatcher(ctx, capability, remoteDON, newTargetFn)
+			if err != nil {
+				return fmt.Errorf("failed to add target shim: %w", err)
+			}
+		default:
+			s.lggr.Warnf("unknown capability type, skipping configuration: %+v", capability)
+		}
+	}
+	return nil
+}
+
+type capabilityService interface {
+	capabilities.BaseCapability
+	remotetypes.Receiver
+	services.Service
+}
+
+func (s *registrySyncer) addToRegistryAndSetDispatcher(ctx context.Context, capabilityInfo kcr.CapabilityRegistryCapability, don kcr.CapabilityRegistryDONInfo, newCapFn func(info capabilities.CapabilityInfo) (capabilityService, error)) error {
+	fullCapID := fmt.Sprintf("%s@%s", capabilityInfo.LabelledName, capabilityInfo.Version)
+	info, err := capabilities.NewRemoteCapabilityInfo(
+		fullCapID,
+		toCapabilityType(capabilityInfo.CapabilityType),
+		fmt.Sprintf("Remote Capability for %s", fullCapID),
+		toDONInfo(don),
+	)
+	if err != nil {
+		return err
+	}
+	capability, err := newCapFn(info)
+	if err != nil {
+		return err
+	}
+
+	err = s.registry.Add(ctx, capability)
+	if err != nil {
+		// If the capability already exists, we've handled this in
+		// a previous syncer iteration, let's skip and move on
+		// to other capabilities.
+		if errors.Is(err, ErrCapabilityAlreadyExists) {
+			return nil
+		}
+
+		return err
+	}
+
+	err = s.dispatcher.SetReceiver(
+		fullCapID,
+		fmt.Sprint(don.Id),
+		capability,
+	)
+	if err != nil {
+		return err
+	}
+	err = capability.Start(ctx)
+	if err != nil {
+		return err
+	}
+	s.subServices = append(s.subServices, capability)
+	return nil
+}
+
+var (
+	defaultTargetRequestTimeout = time.Minute
+)
+
+func (s *registrySyncer) enableExternalAccess(ctx context.Context, myPeerID p2ptypes.PeerID, don kcr.CapabilityRegistryDONInfo, state state, remoteWorkflowDONs []kcr.CapabilityRegistryDONInfo) error {
+	idsToDONs := map[string]capabilities.DON{}
+	for _, d := range remoteWorkflowDONs {
+		idsToDONs[fmt.Sprint(d.Id)] = *toDONInfo(d)
+	}
+
+	for _, c := range don.CapabilityConfigurations {
+		capability, ok := state.IDsToCapabilities[c.CapabilityId]
+		if !ok {
+			return fmt.Errorf("could not find capability matching id %s", c.CapabilityId)
+		}
+
+		switch toCapabilityType(capability.CapabilityType) {
+		case capabilities.CapabilityTypeTrigger:
+			newTriggerPublisher := func(capability capabilities.BaseCapability, info capabilities.CapabilityInfo) (receiverService, error) {
+				cfg := &remotetypes.RemoteTriggerConfig{}
+				cfg.ApplyDefaults()
+				err := proto.Unmarshal(c.Config, cfg)
+				if err != nil {
+					return nil, err
+				}
+				publisher := remote.NewTriggerPublisher(
+					cfg,
+					capability.(capabilities.TriggerCapability),
+					info,
+					*toDONInfo(don),
+					idsToDONs,
+					s.dispatcher,
+					s.lggr,
+				)
+				return publisher, nil
+			}
+
+			err := s.addReceiver(ctx, capability, don, newTriggerPublisher)
```

Comment:
Follow-up: we should limit access to Workflow DON nodes (Peer.IDs). Furthermore, we should implement some rate-limiting as a security measure against a malicious Node DOS'ing the service.

Reply:
Agreed, this should definitely be a follow up